src/classes/GDPR.php

```diff
@@ -416,6 +416,10 @@ public function options_page() {
 		 * @return null
 		 */
 		public function scripts() {
+			$consent = get_option( 'woocommerce_allow_tracking', 'no' );
+			if ( 'no' !== $consent ) {
+				return;
+			}
```

Comment:
We can also move this to the `constructor` function and prevent both the hooks from being added all together.